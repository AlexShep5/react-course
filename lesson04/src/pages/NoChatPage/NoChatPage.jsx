import { Container, Col, Row } from 'react-bootstrap'
import { ChatList } from '../../components/ChatList/ChatList'
import { AddChatForm } from '../../components/AddChatForm/AddChatForm'
import styles from './NoChatPage.module.scss'

export function NoChatPage(props) {

  const addChat = (chat) => {
    const existName = props.chats.find(item => item.name == chat.name)

    if (existName) {
      alert('Чат с таким именем уже существует')
    } else {
      const chatsSize = props.chats.length
      if (chatsSize) {
        chat.id = props.chats[chatsSize - 1].id + 1
      } else {
        chat.id = 1
      }

      const newChats = props.chats
      newChats.push(chat)
      props.setChats(Object.assign([], newChats))
    }
  }

  const delChat = (chatId) => {
    const newChats = props.chats
    const delIndex = newChats.findIndex((item) => item.id == chatId)
    const confDel = window.confirm(`Вы действительно хотите удалить ${newChats[delIndex].name}`)
    if (!confDel) {
      return
    } else {
      newChats.splice(delIndex, 1)
      props.setChats(Object.assign([], newChats))
    }

  }

  return (
    <Container>
      <Row className="mt-4">
        <Col className="col-12 col-sm-6 col-md-6 col-lg-5">
          <h1>Select chat</h1>
          <ChatList className={styles.chatList} chats={props.chats} delChat={delChat}></ChatList>
        </Col>
        <Col className="d-none d-lg-block col-lg-2">
        </Col>
        <Col className="col-12 col-sm-6 col-md-6 col-lg-5 mt-3 mt-sm-0">
          <h1>Add chat</h1>
          <AddChatForm className={styles.userForm} addChat={addChat}></AddChatForm>
        </Col>
      </Row>
    </Container>
  )
}
