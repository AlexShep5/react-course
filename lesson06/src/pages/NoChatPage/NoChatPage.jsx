import { Container, Col, Row } from 'react-bootstrap'
import { ChatList } from '../../components/ChatList/ChatList'
import { AddChatForm } from '../../components/AddChatForm/AddChatForm'
import styles from './NoChatPage.module.scss'

export function NoChatPage() {

  return (
    <Container>
      <Row className="mt-4">
        <Col className="col-12 col-sm-6 col-md-6 col-lg-5">
          <h1>Select chat</h1>
          <ChatList className={styles.chatList}></ChatList>
        </Col>
        <Col className="d-none d-lg-block col-lg-2">
        </Col>
        <Col className="col-12 col-sm-6 col-md-6 col-lg-5 mt-3 mt-sm-0">
          <h1>Add chat</h1>
          <AddChatForm className={styles.userForm}></AddChatForm>
        </Col>
      </Row>
    </Container>
  )
}
