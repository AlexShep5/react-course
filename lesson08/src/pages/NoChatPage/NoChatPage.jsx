import styles from './NoChatPage.module.scss'
import { Container, Col, Row } from 'react-bootstrap'
import { ChatList } from '../../components/ChatList/ChatList'
import { AddChatForm } from '../../components/AddChatForm/AddChatForm'
import { addChat } from '../../store/chats/actions'
import { useSelector, useDispatch } from 'react-redux'
import { getChatList } from '../../store/chats/selectors'
import { deleteChat } from '../../store/chats/actions'

export function NoChatPage() {
  const chatList = useSelector(getChatList)
  const dispatch = useDispatch()

  const onAddChat = (chat) => {
    const existName = chatList.find(item => item.name == chat.name)

    if (existName) {
      alert('Чат с таким именем уже существует')
    } else {
      const chatsSize = chatList.length
      if (chatsSize) {
        chat.id = chatList[chatsSize - 1].id + 1
      } else {
        chat.id = 1
      }
      dispatch(addChat(chat))
    }
  }

  const onDeleteChat = (chatId) => {
    const delName = chatList.find((item) => item.id == chatId).name
    const confDel = window.confirm(`Вы действительно хотите удалить ${delName}`)
    if (confDel) {
      dispatch(deleteChat({id: chatId}))
    }
  }

  return (
    <Container>
      <Row className="mt-4">
        <Col className="col-12 col-sm-6 col-md-6 col-lg-5">
          <h1>Select chat</h1>
          <ChatList className={styles.chatList} chatList={chatList} deletChat={onDeleteChat}></ChatList>
        </Col>
        <Col className="d-none d-lg-block col-lg-2">
        </Col>
        <Col className="col-12 col-sm-6 col-md-6 col-lg-5 mt-3 mt-sm-0">
          <h1>Add chat</h1>
          <AddChatForm className={styles.userForm} addChat={onAddChat}></AddChatForm>
        </Col>
      </Row>
    </Container>
  )
}
