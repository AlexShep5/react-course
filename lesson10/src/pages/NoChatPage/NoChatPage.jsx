import { useEffect } from 'react'
import styles from './NoChatPage.module.scss'
import { Container, Col, Row } from 'react-bootstrap'
import { ChatList } from '../../components/ChatList/ChatList'
import { AddChatForm } from '../../components/AddChatForm/AddChatForm'
import { addChat } from '../../store/chats/actions'
import { useDispatch } from 'react-redux'
import { deleteChat } from '../../store/chats/actions'

export function NoChatPage(props) {
  const dispatch = useDispatch()

  useEffect(() => {
    props.updateDB()
  }, [])

  const onAddChat = (chat) => {
    const existName = props.chats.find(item => item.name == chat.name)

    if (existName) {
      alert('Чат с таким именем уже существует')
    } else {
      const chatsSize = props.chats.length
      if (chatsSize) {
        chat.id = props.chats.reduce((max, item) => {
          if (item.id > max) return item.id
          else return max
        }, 0) + 1
      } else {
        chat.id = 1
      }
      dispatch(addChat(chat))
    }
  }

  const onDeleteChat = (chatId) => {
    const delName = props.chats.find((item) => item.id == chatId).name
    const confDel = window.confirm(`Вы действительно хотите удалить ${delName}`)
    if (confDel) {
      dispatch(deleteChat(delName))
    }
  }

  return (
    <Container>
      <Row className="mt-4">
        <Col className="col-12 col-sm-6 col-md-6 col-lg-5">
          <h1>Select chat</h1>
          <ChatList className={styles.chatList} chatList={props.chats} deletChat={onDeleteChat}></ChatList>
        </Col>
        <Col className="d-none d-lg-block col-lg-2">
        </Col>
        <Col className="col-12 col-sm-6 col-md-6 col-lg-5 mt-3 mt-sm-0">
          <h1>Add chat</h1>
          <AddChatForm className={styles.userForm} addChat={onAddChat}></AddChatForm>
        </Col>
      </Row>
    </Container>
  )
}
