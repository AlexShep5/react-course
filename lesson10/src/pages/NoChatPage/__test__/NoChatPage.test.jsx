import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { NoChatPage } from '..//NoChatPage'
import { Provider } from 'react-redux'
import { store } from '../../../store'
import { BrowserRouter as Router } from 'react-router-dom'
import userEvent from '@testing-library/user-event'


describe('testing ChatsPage', () => {
  const chatList = [
    {
      id: 1,
      name: 'Chat1',
      messages: [{
        text: 'text',
        author: 'author'
      },
      {
        text: 'text',
        author: 'author'
      }]
    },
    {
      id: 2,
      name: 'Chat2',
      messages: [{
        text: 'text',
        author: 'author'
      }]
    }
  ]

  const mockHandler = jest.fn()

  it('render component App', () => {
    render(
      <Provider store={store}>
        <NoChatPage updateDB={() => null}></NoChatPage>
      </Provider>
    )
  })

  it('render with snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <NoChatPage updateDB={() => null}></NoChatPage>
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('render Chat1', () => {
    render(
      <Provider store={store}>
        <Router>
        <NoChatPage chats={chatList} updateDB={() => null}></NoChatPage>
        </Router>
      </Provider>
    )
    expect(screen.getByText('Chat1')).toBeInTheDocument()
  })

  it('render Chat2', () => {
    render(
      <Provider store={store}>
        <Router>
        <NoChatPage chats={chatList} updateDB={() => null}></NoChatPage>
        </Router>
      </Provider>
    )
    expect(screen.getByText('Chat2')).toBeInTheDocument()
  })

})
