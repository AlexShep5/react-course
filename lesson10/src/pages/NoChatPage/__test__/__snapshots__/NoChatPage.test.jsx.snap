// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ChatsPage render with snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="mt-4 row"
    >
      <div
        class="col-12 col-sm-6 col-md-6 col-lg-5 col"
      >
        <h1>
          Select chat
        </h1>
        <div
          class="wrap chatList"
        >
          <div
            class="title"
          >
            Chats
          </div>
        </div>
      </div>
      <div
        class="d-none d-lg-block col-lg-2 col"
      />
      <div
        class="col-12 col-sm-6 col-md-6 col-lg-5 mt-3 mt-sm-0 col"
      >
        <h1>
          Add chat
        </h1>
        <div
          class="wrap userForm"
        >
          <div
            class="MuiFormControl-root MuiTextField-root name css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r2:"
              id=":r2:-label"
            >
              Имя чата
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r2:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Имя чата
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root message css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r3:"
              id=":r3:-label"
            >
              Начальное сообщение
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r3:"
                rows="4"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Начальное сообщение
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="btnWrap"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium button css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Добавить
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
